src/Runner.Worker/StepsRunner.cs

```diff
@@ -352,7 +364,70 @@ private void CompleteStep(IStep step, TaskResult? result = null, string resultCo
         {
             var executionContext = step.ExecutionContext;
 
+            if (ShouldUploadAttachment(step))
+            {
+                QueueStepSummaryUpload(step);
+            }
+
             executionContext.Complete(result, resultCode: resultCode);
         }
+
+        private bool ShouldUploadAttachment(IStep step)
+        {
+          var executionContext = step.ExecutionContext;
+          var stepSummaryFilePath = executionContext.GetGitHubContext("step_summary");
+
+          if(!ActionCommandManager.StepSummaryEnabled(step.ExecutionContext))
+          {
+              Trace.Info("Step Summary is disabled; skipping attachment upload");
+              return false;
+          }
+
+          var summaryExists = File.Exists(stepSummaryFilePath);
+          if (summaryExists)
+          {
+              Trace.Info($"Summary file ({stepSummaryFilePath}) does not exist; skipping attachment upload");
+              return false;
+          }
+          Trace.Info($"Summary file exists: {stepSummaryFilePath}");
+
+          var summaryFileIsEmpty = new FileInfo(stepSummaryFilePath).Length == 0;
+          if (summaryFileIsEmpty)
+          {
+              Trace.Info($"Summary file ({stepSummaryFilePath}) is empty; skipping attachment upload");
+              return false;
+          }
+
+          return true;
+        }
+
+        private string CreateStepSummaryFile(IStep step)
+        {
+            var stepSummaryDirectory = HostContext.GetDirectory(WellKnownDirectory.StepSummary);
```

Comment:
Something to consider here, if the runner does not have permission to create this file, or we run out of disk space, should we try catch here so we don't fail the job for failing to upload this data?

Reply:
added in 901a406, lemme know if we want to be more verbose with specific exceptions!